ui/src/router/components/form/components/ensembler_config/typeOptions.js

```diff
@@ -62,11 +75,10 @@ export const ensemblerTypeOptions = (engineProps) => {
     return typeOptions.filter((o) => o.value !== "standard");
   }
   // Ensembler must be selected when there is an experiment engine
-  const ensemblerOptions = typeOptions.filter((o) => o.value !== "nop");
-  if (
-    !engineProps?.standard_experiment_manager_config
-      ?.experiment_selection_enabled
-  ) {
+  const ensemblerOptions = typeOptions.filter(
+    (o) => o.value !== "nop" && (o.expenginetype === engineProps.type || o.expenginetype === undefined)
+  );
+  if (engineProps?.standard_experiment_manager_config?.experiment_selection_enabled === false) {
```

Comment:
Would `!(engineProps?.standard_experiment_manager_config?.experiment_selection_enabled)` suffice?

Reply:
Nope, a custom experiment engine would not have the `standard_experiment_manager_config` field set, which will cause the condition `!(engineProps?.standard_experiment_manager_config?.experiment_selection_enabled)` to evaluate as `!(undefined)`  which gives `true`, causing the standard ensembler to be removed? 🤔 It isn't much different from the original `(!engineProps?.standard_experiment_manager_config?.experiment_selection_enabled)` right?